File: SpringAngularIntegration/src/main/java/Controller/OrderController.java
package Controller;

import orders.Order;
import orders.OrdersFactory;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/order")
public class OrderController {

    @RequestMapping(method = RequestMethod.GET, value = "/{orderId}/")
    public List<Order> getClient(@PathVariable Long orderId) {
        return OrdersFactory.getInstance().getOrderDAO().getOrdersById(orderId);
    }

    @RequestMapping(method = RequestMethod.GET, value = "/")
    public List<Order> getAllCards(){
        List<Order> orders = OrdersFactory.getInstance().getOrderDAO().getAllOrders();
        return orders;
    }

    @RequestMapping(method = RequestMethod.DELETE, value = "/{orderId}")
    public void delete(@PathVariable Long orderId){
        Order orders = OrdersFactory.getInstance().getOrderDAO().getOrderById(orderId);
        OrdersFactory.getInstance().getOrderDAO().delete(orders);
    }

    @RequestMapping(method = RequestMethod.POST, value = "/")
    public void addClient(@RequestBody Order order){
        OrdersFactory.getInstance().getOrderDAO().add(order);
    }

    @RequestMapping(method = RequestMethod.POST, value = "/{orderId}")
    public void update(@RequestBody Order order, @PathVariable Long orderId){
        order.setId(orderId);
        OrdersFactory.getInstance().getOrderDAO().update(order);
    }
}

File: SpringAngularIntegration/src/main/frontend/controllers/order-controller.js
(function() {
    'use strict';

    angular.module('order.Controller', [])
        .controller('orderController', ['$http', '$routeParams',  function ($http, $routeParams) {
            var vm = this;

            vm.ord = "";

            var orderId = $routeParams.orderId;

            console.log(orderId);

            getAllOrders(orderId);

            function getOrder(id){
                $http.get('http://localhost:8080/order/' + id + '/').then(function(data){
                    vm.order = data.data;
                })
            }

            vm.remove = function(id){
                $http.delete('http://localhost:8080/order/' + id).then(function(){
                    console.log("removing");
                    getAllOrders(orderId);
                })
            };

            vm.add = function(order){
                order.orderId = orderId;
                $http.post('http://localhost:8080/order/', order).then(function(){
                    console.log("adding");
                    console.log(order);
                    getAllOrders(orderId);
                })
            };

            vm.update = function(order){
                $http.post('http://localhost:8080/order/' + order.id, order).then(function(){
                    console.log("updating");
                    console.log(order);
                    getAllOrders(orderId);
                })
            };

            vm.select = function(order){
                vm.ord = order;
                console.log(vm.ord);
            };

            function getAllOrders(orderId){
                $http.get('http://localhost:8080/order/' + orderId + '/').then(function(data){
                    vm.orders = data.data;
                })
            }

        }]);
})();